sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -535,7 +569,13 @@ def _get_labels(
             dtype=compute_dtype,
         )
 
-        X_t, _, centroids_t = self._load_transposed_data_to_device(X, None, centers)
+        X_t, sample_weight, centroids_t = self._load_transposed_data_to_device(
+            X,
+            None
+            if (sample_weight is _IgnoreSampleWeight) and not with_inertia
+            else sample_weight,
```

Comment:
For readability I would rather use an extra local variable instead of using a multi-line ternary operator.